scripts/core/kernel.yml

```diff
@@ -52,7 +49,24 @@ params:
       desc: "[in] size of argument type"
     - type: "const void*"
       name: pArgValue
-      desc: "[in][optional] argument value represented as matching arg type. If null then argument value is considered null."
+      desc: "[in] argument value represented as matching arg type."
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Set kernel argument to a local buffer."
+class: $xKernel
+name: SetArgLocal
+details:
+    - "The implementation of this function should be lock-free."
+params:
+    - type: "$x_kernel_handle_t"
+      name: hKernel
+      desc: "[in] handle of the kernel object"
+    - type: "uint32_t"
+      name: argIndex
+      desc: "[in] argument index in range [0, num args - 1]"
+    - type: "size_t"
+      name: argSize
+      desc: "[in] size of the local buffer to be allocated by the runtime"
```

Comment:
Maybe state the error codes that it returns:

```yml
returns:
        - $X_ERROR_INVALID_KERNEL_ARGUMENT_INDEX
        - $X_ERROR_INVALID_KERNEL_ARGUMENT_SIZE
```

Reply:
Yeah good point, these are missing from the other APIs too. Although this has opened questions about what should be allowed to be null.

Reply:
Added these.